packages/icons/generateIcons.js

```diff
@@ -0,0 +1,124 @@
+const fs = require('fs');
+const path = require('path');
+const prettier = require('prettier');
+const { optimize } = require('svgo');
+
+const iconAssetsDir = './assets/icons';
+const flagAssetsDir = './assets/flags';
+const cryptoIconAssetsDir = './assets/cryptoIcons';
+const iconVariantsFilePath = './src/iconVariants.ts';
+
+// https://github.com/svg/svgo#built-in-plugins
+const svgoConfig = {
+    multipass: true,
+    js2svg: {
+        indent: 2, // string with spaces or number of spaces. 4 by default
+        pretty: true, // boolean, false by default
+    },
+    plugins: [
+        {
+            name: 'preset-default',
+            params: {
+                overrides: {
+                    removeViewBox: false,
+                },
+            },
+        },
+        'removeXMLNS',
+        'prefixIds',
+        // it's necessary to remove all dimension tags to allow resizing
+        'removeDimensions',
+        'removeRasterImages',
+        'removeScriptElement',
+    ],
+};
+
+const optimizeSvgAssets = assetsDir => {
+    const assetFileNames = fs.readdirSync(assetsDir);
+    return assetFileNames
+        .map(fileName => ({
+            fileName,
+            content: fs.readFileSync(path.resolve(assetsDir, fileName)).toString(),
+        }))
+        .map(({ fileName, content }) => ({
+            fileName,
+            content: optimize(content, svgoConfig).data,
+        }));
+};
+
+const writeOptimizedAssets = assetTypesArray => {
+    assetTypesArray.forEach(({ assets, directory }) => {
+        assets.forEach(({ fileName, content }) =>
+            fs.writeFileSync(path.resolve(directory, fileName), content),
+        );
+    });
+};
+
+(async () => {
+    const icons = optimizeSvgAssets(iconAssetsDir);
+    const cryptoIcons = optimizeSvgAssets(cryptoIconAssetsDir);
+    const flagIcons = optimizeSvgAssets(flagAssetsDir);
+
+    const iconVariantsFileContent = `
+    /* eslint-disable global-require */
+
+    // !!! IMPORTANT: This file is autogenerated !!!
+    // If you want to add of modify icons please read README.md to find out how to do it
+
+    export const icons = {
+        ${icons
```

Comment:
what about function for this repeated part?

Reply:
You are right... improved in https://github.com/trezor/trezor-suite/pull/5511/commits/fe1953b6c110fad233fa5a7ee9f18c9268594229